source/grid/grid_refinement.cc

```diff
@@ -515,26 +466,29 @@ GridRefinement::refine_and_coarsen_optimize (Triangulation<dim,spacedim> &tria,
           ExcDimensionMismatch(criteria.size(), tria.n_active_cells()));
   Assert (criteria.is_non_negative (), ExcNegativeCriteria());
 
-  // get an increasing order on
-  // the error indicator
-  std::vector<unsigned int> tmp(criteria.size());
-  for (unsigned int i=0; i<criteria.size(); ++i)
-    tmp[i] = i;
+  // get a decreasing order on the error indicator
+  std::vector<types::global_dof_index> cell_indices(criteria.size());
+  std::iota(cell_indices.begin(), cell_indices.end(), types::global_dof_index(0));
```

Comment:
Yes, indeed odd because it is a noun (which you ordinarily indicate a variable), not a verb as is commonly used for functions.

It is historic: https://en.wikipedia.org/wiki/Iota (section on "Symbol").